modules/user/src/main/TotpSecret.scala

```diff
@@ -0,0 +1,50 @@
+package lila.user
+
+import java.security.SecureRandom
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+import scala.math.{ pow, BigInt }
+import org.apache.commons.codec.binary.Base32
+
+case class TotpSecret(val secret: Array[Byte]) {
```

Comment:
`val` is superfluous. You could `extend AnyVal` so the wrapping class is eliminated during compilation.